Wrote flight class

justin-obrien-smoothstack opened the pull request:
Replaces #4. Includes changes I wasn't able to include in that PR due to what seems like a GitHub bug.

justin-obrien-smoothstack committed:
Write flight PK class

justin-obrien-smoothstack committed:
Add departure time to flight PK class

Re-generate hashCode, equals, serialVersionUID

justin-obrien-smoothstack committed:
Write flight class

justin-obrien-smoothstack committed:
Add Entity annotation to flight class

justin-obrien-smoothstack committed:
Change field types in flight & flight PK classes

Use wrapper objects instead of primitives where applicable
Use Timestamp instead of LocalDateTime
Maximize range of possible ID numbers & minimize memory use
Redo getters/setters, hashCode/equals, UID

sean-darsie commented on src/main/java/com/ss/training/utopia/counter/entity/Flight.java:
```diff
@@ -17,84 +17,87 @@
 	@Id
 	@Column
-	private int departId, arriveId;
+	private Long departId, arriveId;
```
Why make these longs instead of Integers?

justin-obrien-smoothstack replied:
To support the largest possible range of ID values.

More specifically, Java's Integer and MySQL's INT use the same number of bits for integers, but Java's are signed and MySQL's are unsigned, so the maximum value in Java is half that in MySQL.

justin-obrien-smoothstack committed:
Mark flight ID column as unique

Hopefully this will allow the DAO's find by flight ID method to return
a single flight instead of a list

justin-obrien-smoothstack committed:
Add constructors to flight PK class

justin-obrien-smoothstack committed:
Merge branch 'dev' into flight

justin-obrien-smoothstack committed:
Add constructors to flight class

trevorhuis-smoothstack reviewed (approved):
Good Entity
